1: Add IR driver and allow to creat nodes of /dev/input/eventX, and fix gpio_get_value() bug
2: Modify IR driver


diff --git a/drivers/media/rc/gpio-ir-recv.c b/drivers/media/rc/gpio-ir-recv.c
index 80c611c..c051b91 100644
--- a/drivers/media/rc/gpio-ir-recv.c
+++ b/drivers/media/rc/gpio-ir-recv.c
@@ -86,6 +86,10 @@ static irqreturn_t gpio_ir_recv_irq(int irq, void *dev_id)
 	if (gpio_dev->active_low)
 		gval = !gval;
 
+#ifdef CONFIG_ARCH_ADVANTECH
+	(gval > 0)? (gval = 1): (gval = 0);
+#endif
+
 	if (gval == 1)
 		type = IR_PULSE;
 
-- 
1.7.9.5

